### PR TITLE
Yash positive

### PR DESCRIPTION
**Pull request by devyashh**
test pr

### CODE REVIEW

File: src/components/deletePopup.js
Comment (NikhilleadingDots) [inline src/components/deletePopup.js:1]: Wrong way to Delete Please check the correct way how to reuse this component to Delete

File: src/pages/role/roleAdd.js [hunk 1]
```diff
@@ -1,10 +1,61 @@
...
+
+
+    // const handleFileChange = (e) => {
+    //     setRoleData((prevroleData) => ({
+    //         ...prevroleData,
+    //         fileName: e.target.value
+    //     }));
+    // }
+
+    // const handelSubmit = (e) => {
```
Comment (NikhilleadingDots) [inline src/pages/role/roleAdd.js:154]: Review this Function No need to create else if create handleSubmit based on isEdit or not

File: src/pages/role/roleAdd.js [hunk 2]
```diff
@@ -18,15 +69,15 @@ const RoleAdd = () => {
                 </div>
                 <div className='card p-8 '>
                     <div className="formArea mt-5">
-                        <form id="brandAdd">
+                        <form id="roleAdd">
                                 <div className=" w-full ">
                                     <div className="form-group mb-6 ">
                                         <label className="formBlock mb-2 py-2">Name *</label>
                                         <input
```
Comment (NikhilleadingDots) [inline src/pages/role/roleAdd.js:188]: Why there is not a reused Input

File: src/pages/role/roleAdd.js [hunk 3]
```diff
@@ -51,6 +102,7 @@ const RoleAdd = () => {
                                     <button
```
Comment (NikhilleadingDots) [inline src/pages/role/roleAdd.js:218]: Not ANTD button used

**Review by NikhilleadingDots — COMMENTED** [review]
Please check Your Code

File: src/pages/experienceMain/experienceList.js [hunk 4]
```diff
@@ -16,57 +16,17 @@ const ExperienceList = () => {
...
-                    <span className="fa fa-search" /> Search
-                  </button>   
-              </Col>            
-              <Col xs={24} md={6} xl={5}>
-              <div className="btnRight">
+          <h2 className="mb-4"> Experiences</h2>
+          <div className="flex justify-between">
+            <SearchField placeholder="Search by name" />
+            <div className="btnRight">
               <AddButton linkHref="/experience/add" btnText="Add Experience " />
```
Comment (NikhilleadingDots) [inline src/pages/experienceMain/experienceList.js:25]: linkHref No need to use this use onclick or navigate to navigate between tabs

File: src/commonComponents/tableData.js [hunk 5]
```diff
@@ -1185,39 +1185,39 @@ export const dataSource = [
...
-      Distributor: 'Jonas',
-      Email: 'supplier2@yopmail.com',
-      Actions: <>
-      <div className="flex gap-1">
-      <Link to="/distributor/add" className="actionButton"><i className="fa fa-pencil"/></Link>
-      <Link to="/distributor/view" className="actionButton"><i className="fa fa-eye"/></Link>
-      <DeletePopup/>
-      </div>
-      </>,   
-    },
```
Comment (NikhilleadingDots) [inline src/commonComponents/tableData.js:1295]: Remove unused code no need of commented code if you need code Please save nad store it to some backup file at your device

File: src/redux/distributorSlice/distributorSlice.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable */
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+    
+};
```
Comment (NikhilleadingDots) [inline src/redux/distributorSlice/distributorSlice.js:6]: Please add properties to initial state what ever property you need to use in slice

File: src/redux/distributorSlice/distributorSlice.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+export const distributorSlice = createSlice({
+  name: "distributorSlice",
+  initialState,
+  reducers: {
+  }
+});
+
+export const { } = distributorSlice.actions;
+export default distributorSlice.reducer;
```
Comment (NikhilleadingDots) [inline src/redux/distributorSlice/distributorSlice.js:16]: Dont create EmptySlices

File: src/services/distributorApi.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+          url: `/sDUser/lists?role=distributor`,
+          method: "GET"
+        };
+      },
+      providesTags: ["distributorInfo"],
+      async onQueryStarted(arg, { dispatch, getState, queryFulfilled, requestId, extra, getCacheEntry }) {
+        //onStart side-effect
+      try {
+        const { data } = await queryFulfilled;
+        // onSuccess side-effect
```
Comment (NikhilleadingDots) [inline src/services/distributorApi.js:34]: After Sucess Store data in Redux to use it further

File: src/services/supplierApi.js [hunk 9]
```diff
@@ -39,40 +39,40 @@ export const supplierApi = createApi({
...
-    //       // onSuccess side-effect
-    //       notification.success({
-    //         placement: 'top',
-    //         duration: 2,
-    //         closeIcon:true,
-    //         message: 'supplier Created/Updated Successfully',
-    //         description: 'Your supplier has been created/updated successfully.',
-    //         threshold: 2
-    //       });
+    createsupplier: builder.mutation({
```
Comment (NikhilleadingDots) [inline src/services/supplierApi.js:42]: you can create this as Query as you are not mutation any data here

**Review by NikhilleadingDots — COMMENTED** [review]
Please Fix the changes and commit code again

File: src/pages/role/roleAdd.js (hunk 1 above)
Comment (Spideryashh) [reply src/pages/role/roleAdd.js]: updated.

File: src/pages/experienceMain/experienceList.js (hunk 4 above)
Comment (Spideryashh) [reply src/pages/experienceMain/experienceList.js]: changes not done by me.

**Pull request closed without merging**